containers/cpp-mariadb/.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,48 @@
+version: '3.8'
+
+volumes:
+  mariadb-data:
+
+services:
+  app:
+    build: 
+      context: .
+      dockerfile: Dockerfile
+      args:
+        # Update 'VARIANT' to pick a version of CPP
+        # See the README for more information on available versions.
+        VARIANT: debian-11
+    env_file:
+      - .env
+      
+    # Security Opt and cap_add allow for C++ based debuggers to work. 
+    # See `runArgs`: https://github.com/Microsoft/vscode-docs/blob/main/docs/remote/devcontainerjson-reference.md
+    # security_opt:
+    #   - seccomp:unconfined
+    # cap_add:
+    #   - SYS_PTRACE
+
+    volumes:
+      - ..:/workspace:cached
+      
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
+
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:db
+
+    # Comment the next line to use a non-root user for all processes.
+    user: vscode
```

Comment:
Same comment here as in the go definition.  We default the container itself to running as root, but then connect to it using non-root via the remoteUser property. (This is like doing `docker exec -u vscode` after the container has started).  You can then have the entrypoint fire system processes as needed while running as non-root and even uninstalling sudo if you're worried about it.

Reply:
Hi @Chuxel. Fixed with the [latest](https://github.com/microsoft/vscode-dev-containers/pull/1241/commits/92e49336c0ebb4969efbdff5f9902e62a841ee2b) push